content/questions/copying-arrays/index.md

```diff
@@ -0,0 +1,38 @@
+---
+title: Copying Arrays
+tags:
+  - fetch
+  - spread
+order: 32
+date: '2019-10-05'
+answers:
+  - 'A'
+  - 'B'
+  - 'C'
+  - 'A and B'
+  - 'B and C // correct'
+---
+
+How do I make a copy of an array?
+
+```javascript
+const mainArray = ['one', 'two', 'three', 'five', 'four'];
+
+// A
+const a = mainArray;
+
+// B
+const b = [ ...mainArray ];
+
+// C
+const c = [];
+mainArray.forEach( item => {
+  c.push(item);
+});
+```
+
+<!-- explanation -->
+
+Option A just makes a reference to the array. In order to copy all the values from an array, you'll need to iterate through the array and copy each value over, which is what options B and C are doing.
```

Comment:
Recommend adding a note that both B and C are creating shallow copies. Since everything in `mainArray` is a string, this is okay. But, if `mainArray` had objects nested within it, a shallow copy would not copy those nested objects.

Reply:
You got it! Ill take care of this right away :)